hudi-common/src/main/java/org/apache/hudi/internal/schema/convert/AvroInternalSchemaConverter.java

```diff
@@ -70,6 +73,75 @@ public static Schema convert(InternalSchema internalSchema, String name) {
     return buildAvroSchemaFromInternalSchema(internalSchema, name);
   }
 
+  public static InternalSchema pruneAvroSchemaToInternalSchema(Schema schema, InternalSchema originSchema) {
```

Comment:
Let's add a UT for this method separately.

Reply:
Created unit tests for collectColNamesFromSchema. Also created tests to make sure that the spark util creates the same outputs as these tests